### ISSUE TITLE
Displaying of `BindTarget`s seems to be broken

### ISSUE THREAD

**Issue by enricoKoschel** [ItsEthra/egui-bind#1]
The displayed `BindTarget` inside a `Bind` widget is worng for some keys.

Working example:
<img src="https://user-images.githubusercontent.com/10744823/210109315-8955f1ec-d2a4-4af3-b56e-f788ce0e56e0.png" width="400">

Broken example:
<img src="https://user-images.githubusercontent.com/10744823/210109351-72da9354-4595-47ac-af8e-1794b307e710.png" width="400">

In the window you can see from top to bottom the `Bind` widget, the egui `Key` event and the `BindTarget`. Both the `Key` event and the `BindTarget` show the correct key but the `Bind` widget does not.

I've found the widget to be wrong for these keys:
0, 1, 2, 3, 4, 5, 6, 7, +, -

**Comment by enricoKoschel** [comment]
The code for the example above:

`main.rs`
```rust
use eframe::{Frame, NativeOptions};
use egui::{Context, Event, Modifiers, RichText, Widget};
use egui_bind::{Bind, KeyOrPointer};

struct App {
	bind: Option<(KeyOrPointer, Modifiers)>,
	event: Option<Event>,
}

impl eframe::App for App {
	fn update(&mut self, ctx: &Context, frame: &mut Frame) {
		egui::CentralPanel::default().show(ctx, |ui| {
			Bind::new("bind", &mut self.bind).ui(ui);
			ui.separator();

			let e = {
				let input = ui.input();
				input
					.events
					.iter()
					.find(|e| {
						matches!(
							e,
							Event::Key { pressed: true, .. }
								| Event::PointerButton { pressed: true, .. }
						)
					})
					.cloned()
			};

			if let Some(e) = e {
				self.event = Some(e);
			}
			if let Some(e) = &self.event {
				ui.label(RichText::new(format!("egui Event:\n{e:#?}")).size(20.0));
			}

			ui.separator();

			if let Some(bind) = self.bind {
				ui.label(RichText::new(format!("egui-bind BindTarget:\n{bind:#?}")).size(20.0));
			}
		});
	}
}

fn main() {
	eframe::run_native(
		"bind_test",
		NativeOptions::default(),
		Box::new(move |_| {
			Box::new(App {
				bind: None,
				event: None,
			})
		}),
	);
}
```

`cargo.toml`
```
[package]
name = "bind_test"
version = "0.1.0"
edition = "2021"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
egui = "0.20.1"
eframe = "0.20.1"
egui-bind = "0.5.1"
```

**Comment by ItsEthra** [comment]
Thanks, this was due to invalid transmute range to map Key discriminant to integer type. Should be fixed now in `0.5.2`
